### PR TITLE
Add job data model

### PR DESCRIPTION
**Pull request by tkyi**
Job has fields:
- id (number)
- jobName (string)
- pipelineId (number)

routes:
- GET

### CODE REVIEW

**Comment by jer** [comment]
👍

File: test/job.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const validate = require('./helper').validate;
+
+describe('job test', () => {
+    describe('base', () => {
+        it('validates the base', () => {
+            assert.isNull(validate('job.yaml', model.job.base).error);
+        });
+    });
+
+    describe('get', () => {
```
Comment (FenrirUnbound) [inline test/job.test.js]: We test the scenario where the job.yaml is valid and an empty job.yaml. What happens in the scenario where the job.yaml is valid, but has extra fields?

File: model/job.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        .number().positive()
+        .description('Identifier of this Job')
+        .example(52235),
+
+    pipelineId: Joi
+        .number().positive()
+        .description('Identifier of the Pipeline')
+        .example(324),
+
+    jobName: Joi
```
Comment (nkatzman) [inline model/job.js]: does this need to be jobName, or can it just be name?

Comment (tkyi) [inline model/job.js]: name makes more sense given the context. i'll change it

File: test/job.test.js (hunk 1 above)
Comment (tkyi) [inline test/job.test.js]: Added!

Comment (stjohnjohnson) [inline test/job.test.js]: Why would we ever have extra fields?

File: model/job.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+
+const MODEL = {
```
Comment (stjohnjohnson) [inline model/job.js]: Where is the `state`? (enabled, disabled).

 - is it disabled
 - who disabled it
 - when
 - why

Comment (tkyi) [inline model/job.js]: Adding `state` (enabled/disabled) for now. Will address `event_reason` in the future.

File: test/job.test.js (hunk 1 above)
Comment (FenrirUnbound) [inline test/job.test.js]: Because users

Comment (FenrirUnbound) [inline test/job.test.js]: Nevermind. I suppose this is never exposed to the end-user to begin with; it's all internal

**Comment by FenrirUnbound** [comment]
👍 

Seems reasonable to me

krprps8 - imgur

**Comment by jer** [comment]
👍

**Pull request merged**